pkg/definition/gen_sdk/_scaffold/go/pkg/apis/common/application.go

```diff
@@ -0,0 +1,299 @@
+func (a *ApplicationBuilder) GetComponentsByType(typ string) []Component {
+	var result []Component
+	for _, c := range a.components {
+		if c.DefType() == typ {
+			result = append(result, c)
+		}
+	}
+	return result
+}
+
+func (a *ApplicationBuilder) GetWorkflowStepByName(name string) WorkflowStep {
+	for _, s := range a.steps {
+		if s.WorkflowStepName() == name {
+			return s
+		}
+	}
+	return nil
+}
+
+func (a *ApplicationBuilder) GetWorkflowStepsByType(typ string) []WorkflowStep {
+	var result []WorkflowStep
+	for _, s := range a.steps {
+		if s.DefType() == typ {
+			result = append(result, s)
+		}
+	}
+	return result
+}
+
+func (a *ApplicationBuilder) GetPolicyByName(name string) Policy {
+	for _, p := range a.policies {
+		if p.PolicyName() == name {
+			return p
+		}
+	}
+	return nil
+}
+
+func (a *ApplicationBuilder) GetPoliciesByType(typ string) []Policy {
+	var result []Policy
+	for _, p := range a.policies {
+		if p.DefType() == typ {
+			result = append(result, p)
+		}
+	}
+	return result
+}
+
+func (a *ApplicationBuilder) WithWorkflowSteps(step ...WorkflowStep) Application {
+	a.steps = append(a.steps, step...)
+	return a
+}
+
+// WithComponents append components to application
+func (a *ApplicationBuilder) WithComponents(component ...Component) Application {
+	a.components = append(a.components, component...)
+	return a
+}
+
+// WithPolicies append policies to application
+func (a *ApplicationBuilder) WithPolicies(policy ...Policy) Application {
+	a.policies = append(a.policies, policy...)
+	return a
+}
+
+// WithWorkflowMode set the workflow mode of application
+func (a *ApplicationBuilder) WithWorkflowMode(steps, subSteps common.WorkflowMode) Application {
+	a.workflowMode.Steps = steps
+	a.workflowMode.SubSteps = subSteps
+	return a
+}
+
+func (a *ApplicationBuilder) Name(name string) Application {
+	a.name = name
+	return a
+}
+
+func (a *ApplicationBuilder) Namespace(namespace string) Application {
+	a.namespace = namespace
+	return a
+}
+
+func (a *ApplicationBuilder) Labels(labels map[string]string) Application {
+	a.labels = labels
+	return a
+}
+
+func (a *ApplicationBuilder) Annotations(annotations map[string]string) Application {
+	a.annotations = annotations
+	return a
+}
+
+func (a *ApplicationBuilder) GetName() string {
+	return a.name
+}
+
+func (a *ApplicationBuilder) GetNamespace() string {
+	return a.namespace
+}
+
+func (a *ApplicationBuilder) GetLabels() map[string]string {
+	return a.labels
+}
+
+func (a *ApplicationBuilder) GetAnnotations() map[string]string {
+	return a.annotations
+}
+
+// New creates a new application with the given components.
+func New() Application {
+	app := &ApplicationBuilder{
+		components: make([]Component, 0),
+		steps:      make([]WorkflowStep, 0),
+		policies:   make([]Policy, 0),
+	}
+	return app
+}
+
+func (a *ApplicationBuilder) Build() v1beta1.Application {
+	components := make([]common.ApplicationComponent, 0, len(a.components))
+	for _, component := range a.components {
```

Comment:
When user add Trait for K8s-object Component, only Annotations and Labels are valid.
Does `Build()` need validate the compatibility between Component and Trait?

Reply:
This is marked in Future work, thanks for reminding!